### DISCUSSION TITLE
Iterate through request headers

### DISCUSSION THREAD

**Discussion by mfallavol** [uNetworking/uWebSockets#1155] [Q&A] [answered]
Can someone give me an example as to how to iterate through the headers in a request?  I want to gather some information in the .upgrade handler.

I am very proficient with C but not so with the more complicated aspects of C++.  I see that there is a HeaderIterator structure defined in HttpParser.h but I'm just not sure how to use it.

**Comment by VolkerLieber** [comment]
https://unetworking.github.io/uWebSockets.js/generated/interfaces/httprequest.html as a quick help.
I can probably write you something up tomorrow.

**Reply by mfallavol** [reply]
It looks like the forEach functionality only exists in the node.js version.  The C/C++ version doesn't define the forEach method.

**Comment by ghost** [comment, marked as answer]
HttpRequest implements an iterator, so you can just do for (auto header : req)

**Reply by mfallavol** [reply]
Thanks for the quick reply Alex but it doesn't compile.  I get:
    error: 'begin' was not declared in this scope

I thought perhaps it should be
    for (auto header : req->headers)
But the compiler complains that headers is private.

**Reply by ghost** [reply]
for (auto header : *req) maybe

**Reply by mfallavol** [reply]
That did it, thank you very much.
